web3/datastructures.py

```diff
@@ -79,40 +92,42 @@ class AttributeDict(ReadableAttributeDict, Hashable):
     This provides superficial immutability, someone could hack around it
     """
 
-    def __setattr__(self, attr, val):
+    def __setattr__(self, attr: Any, val: Any) -> None:
         if attr == '__dict__':
             super().__setattr__(attr, val)
         else:
             raise TypeError('This data is immutable -- create a copy instead of modifying')
 
-    def __delattr__(self, key):
+    def __delattr__(self, key: Any) -> None:
         raise TypeError('This data is immutable -- create a copy instead of modifying')
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(tuple(sorted(self.items())))
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
         if isinstance(other, Mapping):
             return self.__dict__ == dict(other)
         else:
             return False
 
 
-class NamedElementOnion(Mapping):
+class NamedElementOnion(Mapping[Any, Any]):
```

Comment:
Same here, probably want to define these as `TypeVar` types so that you can specify more concretely what the types of this mapping are (since it is used to house homogenous data and thus can be strongly typed that way)